src/components/ha-form/ha-form-float.ts

```diff
@@ -29,10 +31,12 @@ export class HaFormFloat extends LitElement implements HaFormElement {
         type="numeric"
         inputMode="decimal"
         .label=${this.label}
+        .helper=${this.helper}
+        helperPersistent
         .value=${this.data !== undefined ? this.data : ""}
         .disabled=${this.disabled}
-        .required=${this.schema.required}
-        .autoValidate=${this.schema.required}
+        ?required=${this.schema.required}
+        ?autoValidate=${this.schema.required}
```

Comment:
Why change it to an attribute?

Reply:
Fixing incompatible types

Reply:
I just undid those changes as unrelated and there are a number of other type incompatibilities in those files